[docs] - Improve asset documentation on partitions page (#8355)
dagster-io/dagster: docs/content/concepts/partitions-schedules-sensors/partitions.mdx

---

## Partitioned Assets

### Relevant APIs

| Name                                              | Description                                                                       |
| ------------------------------------------------- | --------------------------------------------------------------------------------- |
| <PyObject object="PartitionsDefinition" />        | Superclass - defines the set of partitions that can be materialized for an asset. |
| <PyObject object="HourlyPartitionsDefinition" />  | A partitions definition with a partition for each hour.                           |
| <PyObject object="DailyPartitionsDefinition" />   | A partitions definition with a partition for each day.                            |
| <PyObject object="WeeklyPartitionsDefinition" />  | A partitions definition with a partition for each week.                           |
| <PyObject object="MonthlyPartitionsDefinition" /> | A partitions definition with a partition for each month.                          |
| <PyObject object="StaticPartitionsDefinition" />  | A partitions definition with a fixed set of partitions.                           |

> A software-defined asset can be assigned a <PyObject object="PartitionsDefinition" />, which determines the set of partitions that compose it. Once an asset has a set of partitions, you can launch materializations of individual partitions, as well as view the materialization history by partition in Dagit.

Reviewer: ```suggestion
A software-defined asset can be assigned a <PyObject object="PartitionsDefinition" /> which determines the set of partitions that compose it. Once an asset has a set of partitions, you can launch materializations of individual partitions and view the materialization history by partition in Dagit.
```

Suggesting we remove an unnecessary comma and make the second sentence more concise

> Here's an asset with a partition for each day since the first day of 2022:

Reviewer: ```suggestion
For example, below is an asset with a partition for each day since the first day of 2022:
```

I feel like introducing examples lends itself to a better overall flow when reading, but this could just be my personal preference. Using words like 'below' to indicate where something is (like an example, in this case) can also be helpful for folks using screen readers.

```python file=/concepts/partitions_schedules_sensors/partitioned_asset.py
from dagster import DailyPartitionsDefinition, asset

@asset(partitions_def=DailyPartitionsDefinition(start_date="2022-01-01"))
def my_daily_partitioned_asset(context):
    context.log.info(
        f"Processing asset partition '{context.asset_partition_key_for_output()}'"
    )
```

When an asset is unpartitioned, the default IO manager stores it in a file whose location is based on the asset's key. When an asset is partitioned, the default IO manager stores each partition in a separate file, all underneath a directory whose location is based on the asset's key.

> If you open up the "Definition" tab of the details page for a partitioned asset, you'll see a bar that represents all of the partitions for the asset. In this case, the bar is entirely gray, because none of the partitions have been materialized.

Reviewer: ```suggestion
To view all partitions for an asset, open the **Definition** tab of the asset's details page. The bar in the **Partitions** section represents all of the partitions for the asset.

In the following image, the partitions bar is entirely gray. This is because none of the partitions have been materialized:
```

Suggesting we break up this paragraph a bit, both for sentence length and skimmability.

<img src="/images/concepts/partitions-schedules-sensors/partitions/partitioned-asset.png" />

> 

Reviewer: Should this content be in a sub-section like `Materializing partitioned assets`?

> When materializing a partitioned asset, you choose which partitions to materialize, and Dagster will launch a run for each partition. If you choose more than one partition, the [Dagster Daemon](/deployment/guides/service#running-dagster-daemon) needs to be running to queue the multiple runs.

Reviewer: ```suggestion
When you materialize a partitioned asset, you'll choose the partitions to materialize and Dagster will launch a run for each partition.

**Note**: If more than one partition is chosen, the [Dagster Daemon](/deployment/guides/service#running-dagster-daemon) must be running to queue the multiple runs.
```

Also breaking this up a bit. It seems like the Dagster daemon sentence is important, so I highlighted it by making it its own sentence and adding `**Note**` to it.

<img src="/images/concepts/partitions-schedules-sensors/partitions/rematerialize-partition.png" />

After you've materialized a partition, it will show up as green in the partitions bar.

<img src="/images/concepts/partitions-schedules-sensors/partitions/materialized-partitioned-asset.png" />

> If you navigate to the "Activity" tab, you'll be able to see materializations by partition:

Reviewer: ```suggestion
To view materializations by partition, navigate to the **Activity** tab:
```

Suggesting we make this more concise and lead with what the outcome is.

<img src="/images/concepts/partitions-schedules-sensors/partitions/materialized-partitioned-asset-activity.png" />

### Partition Dependencies

When a partitioned asset depends on another partitioned asset, each partition in the downstream asset depends on a partition or multiple partitions in the upstream asset.

A few rules govern partition-to-partition dependencies:

- When the upstream asset and downstream asset have the same <PyObject object="PartitionsDefinition" />, each partition in the downstream asset depends on the same partition in the upstream asset.

> - When the upstream asset and downstream asset are both time window-partitioned, each partition in the downstream asset depends on all partitions in the upstream asset that intersect its time window. For example, if an asset with a <PyObject object="DailyPartitionsDefinition" /> depends on an asset with an <PyObject object="HourlyPartitionsDefinition" />, then partition `2022-04-12` of the daily asset the would depend on 24 partitions of the hourly asset: `2022-04-12-00:00` through `2022-04-12-23:00`.

Reviewer: ```suggestion
- When the upstream asset and downstream asset are both time window-partitioned, each partition in the downstream asset depends on all partitions in the upstream asset that intersect its time window.

   For example, if an asset with a <PyObject object="DailyPartitionsDefinition" /> depends on an asset with an <PyObject object="HourlyPartitionsDefinition" />, then partition `2022-04-12` of the daily asset the would depend on 24 partitions of the hourly asset: `2022-04-12-00:00` through `2022-04-12-23:00`.
```

Suggesting we break up this paragraph a bit to make it easier to read.

Author: This example applies to that bullet. I worry that if we broke it up, it would be less clear whether it's an example that references all the rules or just the rule in that bullet.

Author: Ah, I didn't realize it was possible to put multiple paragraphs under a specific bullet. Updating.

Refer to the [Schedules documentation](/concepts/partitions-schedules-sensors/schedules#schedules-from-partitioned-assets-and-jobs) for more info about constructing both schedule types.

## Partitioned Asset Job

> A partitioned job is a job that materializes a particular set of partitioned assets every time it runs.

Reviewer: Should this sentence be `A partitioned asset job is a job...`?

This current iteration strikes me as odd because the previous section is called `Partitioned Jobs`. Based on this sentence I'm not sure what the difference is.

Author: great point. fixed
